isisd/isis_lsp.c

```diff
@@ -1164,6 +1164,26 @@ static void lsp_build(struct isis_lsp *lsp, struct isis_area *area)
 					 false, false, false);
 	}
 
+	/* Check if device should advertise high metrics on startup */
+	if (device_startup) {
+		uint32_t max_metric;
+		max_metric = isis_restart_read_advertise_high_metric(area);
+		if (max_metric > 0) {
+			isis_area_advertise_high_metrics_set(area,
+					  true, max_metric);
+		}
+
+		if (max_metric == ISIS_NARROW_METRIC_INFINITY) {
+			area->oldmetric = 1;
+		} else if (max_metric == MAX_NARROW_LINK_METRIC) {
+			area->oldmetric = 1;
+			area->newmetric = 0;
+		}
```

Comment:
Why are rewritting area->oldmetric and area->newmetric here ?

Reply:
area->oldmetric and area->newmetric determine the metric-style. Based on the value of the max metric we can fetch the configured metric-style upon restart

Reply:
They are already set `isis_area_create`. We cannot overwrite the variable here

Reply:
Don't these set the style of metrics for all the routes? If so, I don't think we should be resetting them here?

Reply:
Removed this block of code, will work after #12819 is merged